### DISCUSSION TITLE
Admin UI : Support reordering of relationship items?

### DISCUSSION THREAD

**Discussion by phil-lgr** [payloadcms/payload#538] [Feature Requests & Ideas]
With the relationship field type: https://payloadcms.com/docs/fields/relationship

We can assign multiple object to it:

![image](https://user-images.githubusercontent.com/9644867/164272950-330ea9fd-0791-462c-9e1b-647dd39383fb.png)

It would be nice to have:

- ability to reorder objects in the list
- allow for a custom preview of each item (instead of the default tag with the name)

I guess my answer would be to create one myself using: https://payloadcms.com/docs/admin/components ?

Thanks!

**Comment by jmikrut** [comment]
Hey @phil-lgr — this is a great request.

Under the hood, we use the fantastic `react-select` package for our Relationship and Select fields. 

It looks like in their docs, there is now an example regarding how to implement reordering / sorting within the input:

https://react-select.com/advanced#sortable-multiselect

This should definitely be built in by default into our existing Relationship and Select fields. I am not sure when we will be able to get to it, but it's a great idea! We would welcome PRs for this function if you'd be up for it. Otherwise we'll certainly slate it in at some point.

When you mention "custom preview", what are you envisioning, exactly? Can you elaborate on what you'd like to see here?

**Comment by jmikrut** [comment]
Hey @phil-lgr — this is now merged into Payload! Just a heads-up.
